common/consistenthash/consistenthash.go

```diff
@@ -21,8 +21,15 @@ func (r *Ring) search(key string) int {
     /////////////////////////
     // YOUR CODE GOES HERE //
     /////////////////////////
-
-    return 0
+	r.Lock()
+	defer r.Unlock()
+	
+	key_hash := hashId(key)
+	var i = 0
+	for i < r.Nodes.Len() && key_hash >= r.Nodes[i].HashId {
```

Comment:
TL;DR: Vai dar certo sempre (ele vai responsabilizar o nó zero), menos pro RemoveNode que é uma função inútil que não deveria existir e não é usada.